src/suites/cts/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,94 @@
+export const description = `
+copy imageBitmap To texture tests.
+`;
+
+import { TestGroup } from '../../framework/index.js';
+
+import { GPUTest } from './gpu_test.js';
+
+export const g = new TestGroup(GPUTest);
+
+function calculateRowPitchForImageBitmap(width: number, bytesPerPixel: number): number {
+  const bytesPerRow = width * bytesPerPixel;
+  return (((bytesPerRow - 1) >> 8) + 1) << 8;
+}
+
+g.test('from image element', async t => {
+  const width = 4;
+  const height = 4;
+  const bytesPerPixel = 4;
+  const imagePixels = new Uint8ClampedArray(bytesPerPixel * width * height);
+  for (let i = 0; i < width * height * bytesPerPixel; ++i) {
+    imagePixels[i] = i % 256;
+  }
+
+  const imageData = new ImageData(imagePixels, width, height);
+  const imageBitmap = await createImageBitmap(imageData);
+
+  //const imageBitmap = await createImageBitmap(img, 0, 0, img.width, img.height);
+  const dst = t.device.createTexture({
+    size: {
+      width: imageBitmap.width,
+      height: imageBitmap.height,
+      depth: 1,
+    },
+    format: 'rgba8uint',
+    usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.COPY_SRC,
+  });
+
+  t.device.defaultQueue.copyImageBitmapToTexture(
+    {
+      imageBitmap,
+      origin: {
+        x: 0,
+        y: 0,
+      },
+    },
+    {
+      texture: dst,
+    },
+    {
+      width: imageBitmap.width,
+      height: imageBitmap.height,
+      depth: 1,
+    }
+  );
+
+  const imageCanvas = document.createElement('canvas');
+  imageCanvas.width = imageBitmap.width;
+  imageCanvas.height = imageBitmap.height;
+
+  const imageCanvasContext = imageCanvas.getContext('2d');
+  t.expect(imageCanvasContext != null);
+  if (!imageCanvasContext) {
+    return;
+  }
+
+  imageCanvasContext.drawImage(imageBitmap, 0, 0, imageBitmap.width, imageBitmap.height);
+  const data = imageCanvasContext.getImageData(0, 0, imageBitmap.width, imageBitmap.height).data;
```

Comment:
can this be removed and just use `imagePixels` as the reference data?

Reply:
I tired using imagePixels directly but failed. And refer to https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/imagebitmap/image_bitmap.cc?type=cs&g=0&l=752 I think createImageBitmap won't keep the imageData as what it is. WDYT?

Reply:
imagePixels itself should definitely not change in value. But you're right, the contents of the ImageBitmap itself might be different (due to premultiplication, and maybe other conversions). It might work if you modify the data generation to avoid invalid values (e.g. just make alpha always 255). 

Otherwise, how about just factoring this out into a helper imageBitmapToData?

Reply:
Make alpha always 255 seems good. And consider the follwing GPU copy path, a helper function seems great. I'm considering the refactor of expectContents by adding default values but don't think it is good so I'll continue on this.
BTW, today our team has some affairs so maybe I cannot update it today. Sry about this.

Reply:
Double check with alpha always 255. It turns out we can check the copy result with imagePixels directly. But still abstract the  drawImage method to cover more cases.